clients/src/test/java/org/apache/kafka/common/record/LazyDownConversionRecordsTest.java

```diff
@@ -160,20 +161,21 @@ private static MemoryRecords convertRecords(MemoryRecords recordsToConvert, byte
                     inputRecords, toMagic, 0L, Time.SYSTEM);
             LazyDownConversionRecordsSend lazySend = lazyRecords.toSend();
             File outputFile = tempFile();
-            FileChannel channel = FileChannel.open(outputFile.toPath(), StandardOpenOption.READ, StandardOpenOption.WRITE);
-
-            int written = 0;
-            while (written < bytesToConvert)
-                written += lazySend.writeTo(channel, written, bytesToConvert - written);
-
-            FileRecords convertedRecords = FileRecords.open(outputFile, true, (int) channel.size(), false);
-            ByteBuffer convertedRecordsBuffer = ByteBuffer.allocate(convertedRecords.sizeInBytes());
-            convertedRecords.readInto(convertedRecordsBuffer, 0);
-
-            // cleanup
-            convertedRecords.close();
-            channel.close();
-
+            ByteBuffer convertedRecordsBuffer;
+            try (FileChannel fileChannel = FileChannel.open(outputFile.toPath(), StandardOpenOption.READ, StandardOpenOption.WRITE)) {
+                ByteBuffer buf;
+                try (org.apache.kafka.common.requests.ByteBufferChannel channel =
+                             new org.apache.kafka.common.requests.ByteBufferChannel(bytesToConvert)) {
```

Comment:
I don't think you can use this as a general class as it stands. A previous PR changed it to be non general, it seems:

```java
    @Override
    public long write(ByteBuffer[] srcs, int offset, int length) {
        int position = buf.position();
        for (int i = 0; i < length; i++) {
            ByteBuffer src = srcs[i].duplicate();
            if (i == 0) {
                src.position(src.position() + offset);
            }
            buf.put(src);
        }
        return buf.position() - position;
    }
```

Reply:
One more thing: why we are using a fully qualified name instead of relying on imports?

Reply:
IIRC, the import rule obstructs me from importing the class. Will check it later.

Reply:
We should fix the underlying case instead of working around it. :)

Reply:
@ijuma Thanks for your comments. I have fixed the method ```write``` and add more tests for it. Also, removing the disallowed package.